java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -353,10 +466,9 @@ class DatabaseImpl(
         fun runSelectQuery(tableName: String) = db.rawQuery(
             "SELECT value FROM $tableName WHERE id = ?",
             arrayOf(valueId.toString())
-        ).use {
-            require(it.moveToFirst()) { "Unknown primitive with ID $valueId." }
-            it.getString(0)
-        }
+        ).forSingleResult { it.getString(0) }
+            ?: throw IllegalArgumentException("Unknown primitive with ID $valueId.")
```

Comment:
Definitely optional: I wonder if it'd be good to add another extension function called `requireSingleResult`?

Reply:
I thought about that, but I couldn't figure out a nice looking way of supplying the lambda and the error message

```
.requireSingleResult("Unknown primitive ...") {
    it.getString(0)
}
```

It seems backwards to me, but the alternative means inlining the lambda, which isn't heaps nice either...

```
.requireSingleResult(
    { it.getString(0) },
    "Unknown primitive ..."
)
```